js/tests/unit/tooltip.js

```diff
@@ -719,6 +719,36 @@ $(function () {
     $styles.remove()
   })
 
+  QUnit.test('should get viewport element from function', function (assert) {
+    assert.expect(2)
+    var styles = '<style>'
+        + '.tooltip, .tooltip .tooltip-inner { width: 200px; height: 200px; max-width: none; }'
+        + '.container-viewport { position: absolute; top: 50px; left: 60px; width: 300px; height: 300px; }'
+        + 'a[rel="tooltip"] { position: fixed; }'
+        + '</style>'
+    var $styles = $(styles).appendTo('head')
+
+    var $container = $('<div class="container-viewport"/>').appendTo(document.body)
+    var $target = $('<a href="#" rel="tooltip" title="tip" style="top: 50px; left: 350px;"/>')
+      .appendTo($container)
+      .bootstrapTooltip({
+        placement: 'bottom',
+        viewport: function ($element) {
+          return ($element.closest('.container-viewport'))
```

Comment:
Perhaps I'm paranoid, but adding an assertion that `$element[0]` is the correct element might be nice.